src/array/_internal/compareValues.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { describe, it, expect } from 'vitest';
+import { compareValues } from './compareValues';
+
+describe('compareValues', () => {
+  it('should return -1 if a < b', () => {
+    expect(compareValues(1, 2, 'asc')).toBe(-1);
+  });
+
+  it('should return 1 if a > b', () => {
+    expect(compareValues(2, 1, 'asc')).toBe(1);
+  });
+
+  it('should return 0 if a === b', () => {
+    expect(compareValues(1, 1, 'asc')).toBe(0);
+  });
+
+  it('should return 1 if a < b and order is desc', () => {
+    expect(compareValues(1, 2, 'desc')).toBe(1);
+  });
+
+  it('should return -1 if a > b and order is desc', () => {
+    expect(compareValues(2, 1, 'desc')).toBe(-1);
+  });
+
+  it('should return 0 if a === b and order is desc', () => {
+    expect(compareValues(1, 1, 'desc')).toBe(0);
+  });
+
+  it('should handle the case where a is string and b is number', () => {
+    const a = ['a', '1b', '3', '2'];
+    const actual = a.map(value => compareValues(value, 1, 'asc'));
+    const expected = a.map(() => 1);
+
+    expect(actual).toEqual(expected);
+  });
```

Comment:
I guess we could improve the test code like this:

```tsx
  it('should compare numbers and strings by converting number into a string', () => {
    expect(compareValues('0', 1, 'asc')).toBe(-1);
    expect(compareValues('a', 1, 'asc')).toBe(1);
    expect(compareValues('1b', 1, 'asc')).toBe(1);
    expect(compareValues('3', 1, 'asc')).toBe(1);
    expect(compareValues('2', 1, 'asc')).toBe(1);
   });
```

I think it is much easier to understand, and it is more comprehensive (since we handle the (`'0'` case)

Reply:
I agree!